gcovr/gcov.py

```diff
@@ -138,30 +141,18 @@ def process_gcov_data(
 
     key = os.path.normpath(fname)
 
-    parser_flags = ParserFlags.NONE
-    if options.gcov_ignore_parse_errors:
-        parser_flags |= ParserFlags.IGNORE_PARSE_ERRORS
-    if options.exclude_function_lines:
-        parser_flags |= ParserFlags.EXCLUDE_FUNCTION_LINES
-    if options.exclude_internal_functions:
-        parser_flags |= ParserFlags.EXCLUDE_INTERNAL_FUNCTIONS
-    if options.exclude_unreachable_branches:
-        parser_flags |= ParserFlags.EXCLUDE_UNREACHABLE_BRANCHES
-    if options.exclude_throw_branches:
-        parser_flags |= ParserFlags.EXCLUDE_THROW_BRANCHES
-    if options.respect_exclusion_markers:
-        parser_flags |= ParserFlags.RESPECT_EXCLUSION_MARKERS
-    if options.show_decision:
-        parser_flags |= ParserFlags.PARSE_DECISIONS
-
-    coverage = parse_coverage(
+    coverage, source_lines = parse_coverage(
         lines,
         filename=key,
-        exclude_lines_by_pattern=options.exclude_lines_by_pattern,
-        exclude_branches_by_pattern=options.exclude_branches_by_pattern,
-        exclude_pattern_prefix=options.exclude_pattern_prefix,
-        flags=parser_flags,
+        ignore_parse_errors=options.gcov_ignore_parse_errors,
     )
+
+    apply_all_exclusions(coverage, lines=source_lines, options=options)
```

Comment:
`ExclusionOptions` acts more like an interface.

The `options` data structure is completely untyped, as it is assembled dynamically by argparse and our configuration handling code. Within the exclusion processing code, `ExclusionOptions` provides some static type checking. But since the code isn't fully typed, this line here is essentially an unchecked cast.

While `options` is not `isinstance(options, ExclusionOptions)`, it does provide all fields expected by the ExclusionOptions.